AppCenterDataStorage/AppCenterDataStorage/MSDataSourceError.h

```diff
@@ -19,4 +24,23 @@
  */
 - (instancetype)initWithError:(NSError *)error;
 
+/**
+ * Create an instance with error object.
+ *
+ * @param error An error object.
+ * @param errorCode An error code.
+ *
+ * @return A new `MSDataSourceError` instance.
+ */
+- (instancetype)initWithError:(NSError *)error errorCode:(NSInteger)errorCode;
+
+/**
+ * Get the error code from the error.
+ *
+ * @param error An error object.
+ *
+ * @return The error code.
+ */
++ (NSInteger)errorCodeWithError:(NSError *)error;
```

Comment:
`errorCodeFromError:error` naming makes more sense to me. What do you think?

Reply:
convention is to provide methodNameWithFirstArgument:

Reply:
@MehrdadMzfr While we stick to that convention most of the time, but in this case, at least it makes sense to me to make it more readable as the return type "errorCode" is something to extract from the first parameter "error".

Anyway, I don't have a strong preference and it's not a big deal either way, just sharing my thoughts.